### PR TITLE
Add CLI function to convert script-dataset to Parquet

### LINKED ISSUE #6690: Add function to convert a script-dataset to Parquet
Add function to convert a script-dataset to Parquet and push it to the Hub, analogously to the Space: "Convert a Hugging Face dataset to Parquet"

### PR DESCRIPTION
**Pull request by albertvillanova**
Close #6690.

### CODE REVIEW

**Comment by albertvillanova** [comment]
@huggingface/datasets once this PR is merged, I would suggest making a release. Do you agree?

File: src/datasets/commands/convert_to_parquet.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        print(f"{configs = }")
+        default_config = get_dataset_default_config_name(dataset_id, token=token, revision=revision)
+        print(f"{default_config = }")
+        if default_config:
+            config = default_config
+            configs.remove(default_config)
+        else:
+            config = configs.pop(0)
+        print(f"{config = }")
+        dataset = load_dataset(dataset_id, config, revision=revision)
```
Comment (lhoestq) [inline src/datasets/commands/convert_to_parquet.py:53]: Maybe add `trust_remote_code` ?

(also in get_dataset_config_names and get_dataset_default_config_name)

**Review by lhoestq — APPROVED** [review]
LGTM :)

File: src/datasets/commands/convert_to_parquet.py (hunk 1 above)
Comment (albertvillanova) [reply src/datasets/commands/convert_to_parquet.py:53]: Yes, otherwise we are raising warning now and eventually raise an error.

Comment (albertvillanova) [reply src/datasets/commands/convert_to_parquet.py]: Done.

File: src/datasets/commands/convert_to_parquet.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            dataset = load_dataset(dataset_id, config, revision=revision, trust_remote_code=trust_remote_code)
+            dataset.push_to_hub(
+                dataset_id,
+                config_name=config,
+                commit_message=f"Add {config} data files",
+                revision=pr_revision,
+                token=token,
+            )
+            time.sleep(5)
+        delete_files(dataset_id, revision=pr_revision, token=token)
```
Comment (severo) [inline src/datasets/commands/convert_to_parquet.py:90]: When providing support, we sometimes suggest that users store their script in a `script` branch. What do you think of this alternative to deleting the files?

TBH, I think it's OK to delete the files and to let the users move their script to another branch manually if they want to. Just wanted to check it's not an oversight

Comment (albertvillanova) [reply src/datasets/commands/convert_to_parquet.py:90]: If we agree this is interesting, I could have a look at it, using the huggingface_hub client `create_branch` function, and see if the logic is easily implementable.

Comment (albertvillanova) [reply src/datasets/commands/convert_to_parquet.py:90]: I agree many users will like to keep the script somewhere for future use/updates.

Comment (albertvillanova) [reply src/datasets/commands/convert_to_parquet.py:90]: Done. Thanks @severo for the proposal.

**Pull request merged**